modules/eventing/pages/eventing-curl-spec.adoc

```diff
@@ -0,0 +1,235 @@
+
+Cookie support may be enabled at binding level if desired when accessing controlled and trusted endpoints.
+
+== Example
+
+In the below example, a cURL request is created to the specified binding profile_svc_binding with the sub-URL /person with URL parameters action and id and the body being a JSON object. The response is a JSON object and is seen containing a field profile_id. In this example, the request is automatically encoded as application/json and response is automatically parsed from JSON response, as no explicit encoding is specified.
+
+----
+var request = {
+    path: '/person',
+    params: {
+        'action': 'create',
+        'id': 23012
+    },
+    body: {
+        'name': 'John Smith',
+        'age': 25,
+        'state': 'CA',
+        'country': 'US',
+    }
+};
+
+var response = curl('POST', profile_svc_binding, request);
+if (response.status == 200) {
+  var profile_id = response.body.profile_id;
+  log("Successfully created profile " + profile_id);
+}
+----
+
+== Request marshalling
+
+
+[#optional-id,cols="1,1,1,1",options="header"]    
+|===
+
+| *JS object passed to the body param*
+| *Value passed for encoding param*
+| *Encoding used for request body*
+| *Content-Type header sent* (unless overridden by headers param)
+
+| 
+| 
+| 
+| 
+
+| JS String
+| (not specified)
+| UTF-8
+| text/plain
+
+| JS Object
+| (not specified)
+| JSON
+| application/json
+
+| JS ArrayBuffer
+| (not specified)
+| Raw Bytes
+| application/octet-stream
+
+| 
+| 
+| 
+| 
+
+| JS String
+| TEXT
+| UTF-8
+| text/plain
+
+| JS Object
+| TEXT
+| (disallowed)
+| (disallowed)
+
+| JS ArrayBuffer
+| TEXT
+| (disallowed)
+| (disallowed)
+
+| 
+| 
+| 
+| 
+
+| JS String
+| FORM
+| URL Encoding
+| application/x-www-form-urlencoded
+
+| JS Object
+| FORM
+| URL Encoding
+| application/x-www-form-urlencoded
+
+| JS ArrayBuffer
+| FORM
+| (disallowed)
+| (disallowed)
+
+| 
+| 
+| 
+| 
+
+| JS String
+| JSON
+| JSON
+| application/json
+
+| JS Object
+| JSON
+| JSON
+| application/json
+
+| JS ArrayBuffer
+| JSON
+| (disallowed)
+| (disallowed)
+
+| 
+| 
+| 
+| 
+
+| JS String
+| BINARY
+| UTF-8
+| application/octet-stream
+
+| JS Object
+| BINARY
+| (disallowed)
+| (disallowed)
+
+| JS ArrayBuffer
+| BINARY
+| Raw Bytes
+| application/octet-stream
+|===
+
+Users who wish to utilize custom encoding can do so by specifying an appropriate Content-Type using the _headers_ parameter of the request object and passing the custom encoded object as an ArrayBuffer as the _body_ parameter of the request.
+
+== Response unmarshalling
+
+Response object from the remote is automatically unmarshalled if the response contains a recognized Content-Type header. The following table identifies the action used to unmarshall responses:
+
+[#optional-id,cols="1,1,1",options="header"]    
```

Comment:
Duplicate id (`optional-id`) for this table. Please either change the id to something unique, or remove the id altogether. You still need to set the `cols` and `options` attributes:

```
[cols="1,1,1",options="header"]
```